lib/Byng/Pimcore/Elasticsearch/Gateway/AbstractGateway.php

```diff
@@ -184,21 +119,25 @@ protected function doSearch(
      *
      * @return array
      */
-    protected function processQuery(Query $query)
+    protected function processQuery(QueryInterface $query)
```

Comment:
This function is the culprit then - when we talked yesterday I was thinking the actual query classes themselves would be able to "render" themselves, so this would become something more like:

```php
$result[$query->getType()] = $query->render();
```

Or something? Maybe that's something that could be a todo for the future to allow any kind of user-designed query to be made by implementing the query interface.